### ISSUE TITLE
Decode always gives nil

### ISSUE THREAD

**Issue by Romaboy** [bungle/lua-resty-libcjson#8]
```moonscript
json = require "resty.libcjson"
json.decode "{}" -- always nil
json.decode "true" -- null, very strange
json.decode "'test'" -- nil
```
While encoding works well. Which installation could I miss?
Thanks!

**Comment by bungle** [comment]
For me it gives:

```lua
table: 0x00053260
true
nil
```

The last `nil` is because JSON does not have single quoted strings, so it is invalid JSON.

**Comment by bungle** [comment]
What libcjson library are you using?

**Comment by bungle** [comment]
Now I used https://github.com/DaveGamble/cJSON, and got (same with latest: https://github.com/kbranigan/cJSON)

```lua
nil
table: 0x00051760
nil
```

Something has changed in libcjson C-library.

**Comment by bungle** [comment]
I think this explains the weirdness:
https://github.com/DaveGamble/cJSON/commit/ce546b54d8fe6afee69763ca78ded3195d8d21e7#diff-2e7217b36cb40e79681119516185f573

**Comment by bungle** [comment]
It is now fixed in this commit:
https://github.com/bungle/lua-resty-libcjson/commit/9b4e4c73f8d61c026b2738ffdb1330bac5ca9498

**Comment by bungle** [comment]
I also released 1.4 that contains the above fix:
https://github.com/bungle/lua-resty-libcjson/releases/tag/v1.4

It is also now on Lua Rocks:
https://luarocks.org/modules/bungle/lua-resty-libcjson

**Comment by Romaboy** [comment]
Awesome, thanks for fixing so fast!
